commit ea0901c231dd2b062597bafb13fd6022e67826ca (refs/changes/36/136/3)
Author: Johanan Lai <lai.johanan@gmail.com>
Date:   2016-07-26T15:29:51-07:00 (3 years, 2 months ago)
    
    ADMINUI-2323 UUID should be validated similar to searching with IP and MAC address parameters

diff --git a/lib/vms.js b/lib/vms.js
index 9600f103..21a3bd95 100644
--- a/lib/vms.js
+++ b/lib/vms.js
@@ -112,10 +112,24 @@ var MAC_REGEX =
 
 function list(req, res, next) {
     var params = req.params;
+    var uuid = params.uuid;
     var ip = params.ip;
     var mac = params.mac;
+
+    // Check if UUID is valid first
+    if (uuid) {
+        if (!UUID_REGEX.test(uuid)) {
+            return next(
+                new errors.InvalidParameterError('Invalid UUID', [ {
+                    field: 'uuid',
+                    code: 'invalid',
+                    message: sprintf('UUID %s is not a valid UUID', uuid)
+                } ]));
+        }
+    }
+    // Now check if there are additional parameters being searched
     if (ip) {
-        if (false === IPV4_REGEX.test(ip)) {
+        if (!IPV4_REGEX.test(ip)) {
             return next(
                 new errors.InvalidParameterError('Invalid IP Address', [ {
                     field: 'ip',
@@ -125,7 +139,7 @@ function list(req, res, next) {
         }
         return listVmsWithIP(ip);
     } else if (mac) {
-        if (false === MAC_REGEX.test(mac)) {
+        if (!MAC_REGEX.test(mac)) {
             return next(
                 new errors.InvalidParameterError('Invalid MAC Address', [ {
                     field: 'mac',
